pkg/streamingpromql/testing.go

```diff
@@ -24,3 +30,142 @@ func NewTestEngineOpts() EngineOpts {
 		Pedantic:       true,
 	}
 }
+
+// Why do we do this rather than require.Equal(t, expected, actual)?
+// It's possible that floating point values are slightly different due to imprecision, but require.Equal doesn't allow us to set an allowable difference.
+func RequireEqualResults(t testing.TB, expr string, expected, actual *promql.Result, checkAnnotations bool) {
```

Comment:
What do you think about moving this to a test utilities package or something like that?

As it stands, this introduces a production dependency on `testify/require`, which feels a bit weird.